### ISSUE TITLE
[HOLD for payment 2024-03-13] Group chat - Join button appears in header after creating a group chat

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37709]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.47-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Start chat.
3. Create a group chat.

## Expected Result:
Join button will not appear in group chat.

## Actual Result:
Join button shows up in group chat header.
The same issue also occurs when User B opens task report created by User A.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/a2babbf3-88ee-483e-b15f-31c524fb388f

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by m-natarajan** [comment]
We think that this bug might be related to #vip-vsb
cc @quinthar

**Comment by m-natarajan** [comment]
@MonilBhavsar FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Group chat - Join button appears in header after creating a group chat
### What is the root cause of that problem?
- Related to https://github.com/Expensify/App/pull/37165
- We are using ```const canJoinOrLeave = isChatThread || !isSelfDM || isUserCreatedPolicyRoom || canLeaveRoom;``` that is ```true``` to display the join button
### What changes do you think we should make in order to solve the problem?
- Should use ```    const canJoinOrLeave = (isChatThread || isUserCreatedPolicyRoom || canLeaveRoom) && !isSelfDM;``` instead

### What alternative solutions did you explore? (Optional)
- NA

**Comment by ShridharGoel** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Join button appears in header after creating a group chat.

### What is the root cause of that problem?
There is a check `!isSelfDM` in `canJoinOrLeave` but no way to check if user is already a participant.

```
 const canJoinOrLeave = isChatThread || !isSelfDM || isUserCreatedPolicyRoom || canLeaveRoom;
```

### What changes do you think we should make in order to solve the problem?
A check needs to be added for ensuring that user is not already a participant while creating `canJoin`.

```
    const canJoin = canJoinOrLeave && !isWhisperAction && props.report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN && !ReportUtils.isReportParticipant(currentUserAccountID ?? 0, props.report);
```

**Comment by ShridharGoel** [comment]
@MonilBhavsar Are we considering proposals for this since the original PR is older than a week?

**Comment by MonilBhavsar** [comment]
> Are we considering proposals for this since the original PR is older than a week?

Not older than week; the PR was deployed to staging yesterday.

**Comment by MonilBhavsar** [comment]
@ishpaul777 and @paultsimura assigned you both to this issue

**Comment by MonilBhavsar** [comment]
This is fixed on staging. Thanks all!

**Comment by paultsimura** [comment]
> The solution for this issue has been 🚀 deployed to production 🚀 in version 1.4.47-10 and is now subject to a 7-day regression period 📆. Here is the list of pull requests that resolve this issue:
> 
> * [[CP staging] Fix join button displayed in group chat #37726](https://github.com/Expensify/App/pull/37726)
> 
> If no regressions arise, payment will be issued on 2024-03-13. 🎊
> 
> For reference, here are some details about the assignees on this issue:
> 
> * @paultsimura requires payment (Needs manual offer from BZ)
> * @ishpaul777 requires payment (Needs manual offer from BZ)

No payment here – it was fixed as a regression.

**Comment by MonilBhavsar** [comment]
No payment required. So closing this. Thanks all!
